wp-content/themes/inf_theme/includes/class-main.php

```diff
@@ -213,50 +193,23 @@ public function run() {
   }
 
   /**
-   * The reference to the class that orchestrates the hooks.
+   * Define global variable to save memory when parsing manifest on every load.
    *
-   * @return Loader Orchestrates the hooks.
-   *
-   * @since 2.0.0
+   * @since 3.0.0
    */
-  public function get_loader() {
-    return $this->loader;
-  }
+  public function set_assets_manifest_data() {
+    $response = wp_remote_get( INF_ASSETS_PUBLIC_URL . 'manifest.json' );
 
-  /**
-   * The name used to uniquely identify it within the context of
-   * WordPress and to define internationalization functionality.
-   *
-   * @return string Theme name.
-   *
-   * @since 2.0.0
-   */
-  public function get_theme_name() {
-    return $this->theme_name;
-  }
+    if ( ! is_array( $response ) && is_wp_error( $response ) ) {
+      return;
+    }
 
-  /**
-   * Retrieve the version number.
-   *
-   * @return string Theme version number.
-   *
-   * @since 2.0.0
-   */
-  public function get_theme_version() {
-    return $this->theme_version;
-  }
+    $parsed_data = json_decode( $response['body'] );
```

Comment:
Passing `true` as a second argument to `json_decode` will transform it to array instead of object.

http://php.net/manual/en/function.json-decode.php